modules/reactor/src/meshgenerators/PeripheralTriangleMeshGenerator.C

```diff
@@ -0,0 +1,470 @@
+  params.addRequiredParam<MeshGeneratorName>(
+      "inner_boundary_mesh", "The mesh to use for the inner boundary of the triangulation.");
+  params.addRangeCheckedParam<boundary_id_type>(
+      "inner_boundary_id",
+      "inner_boundary_id>=0",
+      "The boundary ID of the inner mesh outer boundary.");
+  params.addParam<std::string>("inner_boundary_name",
+                               "The boundary name of the inner mesh outer boundary.");
+  params.addRequiredRangeCheckedParam<Real>(
+      "outer_circle_radius", "outer_circle_radius>0", "Radius of outer circle boundary.");
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "outer_circle_num_segments",
+      "outer_circle_num_segments>0",
+      "Number of radial segments to subdivide outer circle boundary.");
+  params.addRangeCheckedParam<boundary_id_type>(
+      "outer_boundary_id",
+      "outer_boundary_id>=0",
+      "The boundary id for the generated mesh outer boundary.");
+  // params.addParam<std::string>(
+  //   "outer_boundary_name",
+  //   "The boundary name of the generated mesh outer boundary.");
+  params.addParam<std::vector<Real>>("extra_circle_radii", "Radii of extra Steiner point circles.");
+  params.addParam<std::vector<unsigned int>>("extra_circle_num_segments",
+                                             "Number of segments for extra Steiner point circles.");
+  params.addClassDescription("This PeripheralTriangleMeshGenerator object is designed to generate "
+                             "a triangulated mesh between a generated outer circle boundary "
+                             "and a provided inner mesh.");
+
+  return params;
+}
+
+PeripheralTriangleMeshGenerator::PeripheralTriangleMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _tri_subdomain_id(isParamValid("subdomain_id") ? getParam<SubdomainID>("subdomain_id") : 0),
+    _inner_boundary_mesh(getMesh("inner_boundary_mesh")),
+    _outer_circle_radius(getParam<Real>("outer_circle_radius")),
+    _outer_circle_num_segments(getParam<unsigned int>("outer_circle_num_segments")),
+    _outer_boundary_id(
+        isParamValid("outer_boundary_id") ? getParam<boundary_id_type>("outer_boundary_id") : 0),
+    // _outer_boundary_name(isParamValid("outer_boundary_name") ?
+    // getParam<std::string>("outer_boundary_name") : std::string("outside")),
+    _extra_circle_radii(getParam<std::vector<Real>>("extra_circle_radii")),
+    _extra_circle_num_segments(getParam<std::vector<unsigned int>>("extra_circle_num_segments"))
+{
+  // confirm either id or name (exclusive or) for inner mesh boundary is provided
+  if ((!isParamValid("inner_boundary_id") && !isParamValid("inner_boundary_name")) ||
+      (isParamValid("inner_boundary_id") && isParamValid("inner_boundary_name")))
+  {
+    paramError(
+        "inner_boundary_id",
+        "Either 'inner_boundary_id' or 'inner_boundary_name' must be provided, but not both.");
```

Comment:
```suggestion
        "Either 'inner_boundary_id' or 'inner_boundary_name' must be provided.");
```

Reply:
Isn't `but not both` accurate here?  If both are provided then we can't tell which to use.  (I suppose we could check the naming to see if the specifications were consistent and then carry on if so, but IMHO "just pick one or the other" is a reasonable user interface)

Reply:
Yeah that’s fine. That error comes out when none are provided too, but in that case extraneously saying “not both” is okay